src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -159,46 +173,38 @@ void HelpPromptManagerImpl::OnSetGlobalPropertiesReceived(
                   " do not track SetGlobalProperties");
     return;
   }
-  sending_type_ = GetSendingType(msg, is_response);
-  LOG4CXX_DEBUG(logger_, "Set sending type to:" << unsigned(sending_type_));
-}
 
-void HelpPromptManagerImpl::OnAppActivated(const bool is_restore) {
-  LOG4CXX_AUTO_TRACE(logger_);
-  LOG4CXX_DEBUG(logger_, "Activated for appID:" << app_.app_id());
+  if (!is_response) {
+    if (msg.keyExists(strings::help_prompt)) {
+      is_tts_send_ = true;
+    }
+    if (msg.keyExists(strings::vr_help)) {
+      is_ui_send_ = true;
+    }
 
-  if (is_restore) {
-    LOG4CXX_DEBUG(logger_, "restoring");
-    is_tts_send_ = (app_.help_prompt() && app_.help_prompt()->length());
-    is_ui_send_ = (app_.vr_help() && app_.vr_help()->length());
     LOG4CXX_DEBUG(logger_, "is_tts_send_:" << is_tts_send_);
     LOG4CXX_DEBUG(logger_, "is_ui_send_:" << is_ui_send_);
-    if (is_tts_send_ && is_ui_send_) {
-      sending_type_ = SendingType::kNoneSend;
-      LOG4CXX_DEBUG(logger_,
-                    "SetGlobalProperties RPC"
-                    " with the vrHelp and helpPrompt"
-                    " has been sent");
-      return;
-    } else {
-      sending_type_ = is_tts_send_ ? SendingType::kSendVRHelp : sending_type_;
-      sending_type_ =
-          is_ui_send_ ? SendingType::kSendHelpPrompt : sending_type_;
-      is_tts_send_ = false;
-      is_ui_send_ = false;
-      const DataAccessor<CommandsMap> accessor = app_.commands_map();
-      const CommandsMap& commands = accessor.GetData();
-      for (auto& command : commands) {
-        AddCommand(command.first, *command.second);
-      }
-    }
+    return;
   }
-  SendRequests();
+
+  SetSendingType(msg);
 }
 
-void HelpPromptManagerImpl::OnAppUnregistered() {
-  LOG4CXX_AUTO_TRACE(logger_);
-  LOG4CXX_DEBUG(logger_, "Unregisted for appID:" << app_.app_id());
+HelpPromptManagerImpl::SendingType HelpPromptManagerImpl::GetSendingType()
+    const {
+  if (is_tts_send_ && is_ui_send_) {
```

Comment:
Shouldn't this just be `return sending_type_;`, since SetSendingType covers this logic?

Reply:
@jacobkeeler good point. Fixed in 1c57695b83f1d680a8ea23513378e6c6248210da